sdk/eidos_sdk/util/stream_collector.py

```diff
@@ -16,69 +17,80 @@
 
 
 class StreamCollector(AsyncIterator[StreamEvent]):
```

Comment:
Split this apart again

Stream collector now does nothing but aggregate events

stream_manager helper function handles wrapping with context events + error handling